File: CustomCellXcode/CustomCellXcode/ViewController.m
//
//  ViewController.m
//  CustomCellXcode
//
//  Created by Rustam on 27/07/2015.
//  Copyright © 2015 Rustam. All rights reserved.
//

#import "ViewController.h"
#import "MyCustomCell.h"

@interface ViewController ()

@property (strong, nonatomic) NSArray * items;
@end

@implementation ViewController
@synthesize items = _items;

- (NSArray *)items
{
    if (!_items)
    {
        NSMutableArray * arr = [NSMutableArray arrayWithCapacity:20];
        for (NSInteger i=0; i<20; i++)
            [arr addObject:[NSString stringWithFormat:@"Item%d", i]];
        _items = arr;
    }
    return _items;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
{
    return [self.items count];
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
{
    MyCustomCell * cell = [tableView dequeueReusableCellWithIdentifier:@"myCell"];
    if (!cell)
    {
        [tableView registerNib:[UINib nibWithNibName:@"MyCustomCell" bundle:nil] forCellReuseIdentifier:@"myCell"];
        cell = [tableView dequeueReusableCellWithIdentifier:@"myCell"];
    }
    
    return cell;
}

- (void)tableView:(UITableView *)tableView willDisplayCell:(MyCustomCell *)cell forRowAtIndexPath:(NSIndexPath *)indexPath
{
    cell.LeftLbl.text = [self.items objectAtIndex:indexPath.row];
    cell.RightLbl.text = [self.items objectAtIndex:indexPath.row];
    cell.MiddleLbl.text = [self.items objectAtIndex:indexPath.row];
}

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: CustomCellXI/CustomCellXI/ViewController.cs
using System;

using UIKit;
using System.Collections.Generic;

namespace CustomCellXI
{
	public partial class ViewController : UIViewController, IUITableViewDataSource
	{
		List<string> items;

		public ViewController (IntPtr handle) : base (handle)
		{
		}

		public override void ViewDidLoad ()
		{
			base.ViewDidLoad ();
			items = new List<string> ();
			for (int i = 0; i < 20; i++) {
				items.Add (string.Format ("item {0}", i));
			}

			Table.DataSource = this;
			// Perform any additional setup after loading the view, typically from a nib.
		}

		public nint RowsInSection (UITableView tableView, nint section)
		{
			return items.Count;
		}
		public UITableViewCell GetCell (UITableView tableView, Foundation.NSIndexPath indexPath)
		{
			var cell = (CustomCell)tableView.DequeueReusableCell ("myCell");
			if (cell == null) {
				tableView.RegisterNibForCellReuse (UINib.FromName("CustomCell", null), "myCell");
				cell = (CustomCell)tableView.DequeueReusableCell ("myCell");
			}

			return cell;
		}

		public override void DidReceiveMemoryWarning ()
		{
			base.DidReceiveMemoryWarning ();
			// Release any cached data, images, etc that aren't in use.
		}
	}
}

